rust/sealevel/programs/mailbox/src/processor.rs

```diff
@@ -637,15 +645,18 @@ fn outbox_dispatch(
         return Err(ProgramError::from(Error::ExtraneousAccount));
     }
 
-    if dispatch.message_body.len() > MAX_MESSAGE_BODY_BYTES {
-        return Err(ProgramError::from(Error::MaxMessageSizeExceeded));
-    }
-
     let count = outbox
         .tree
         .count()
         .try_into()
         .expect("Too many messages in outbox tree");
+
+    let protocol_fee = outbox.protocol_fee.fee;
+    invoke(
+        &system_instruction::transfer(payer_info.key, outbox_info.key, protocol_fee),
```

Comment:
do you know if this will log that the transfer occurred?

I'm thinking about how we'll want to start indexing protocol fees in the scraper very soon. The IGP payments / messages / message deliveries all use a dedicated account where info about it is written.

I think for protocol fees, our options are:

1. put the protocol fee payment in the dispatched message account that's written
2. have a log using `msg!` that we can use to figure out how much protocol fee was paid in a given tx. Bc of hook indexing, we just need the tx hash and can find this using https://solana.com/docs/rpc/http/gettransaction which shows all the logs for the tx
3. the scraper could maybe just assume that a configured protocol fee was paid

2 feels the most reasonable to me - wdyt? If the system program doesn't already log this, wonder if we should add a `msg!` saying the protocol fee was paid and how much it was

Reply:
I checked several txs we made when deploying / sending warp transfers and couldn't find system transfer logs, so I doubt we can assume that.


Option 2 seems to be working well yep, hook indexing ftw:
```
curl --location 'https://mainnetbeta-rpc.eclipse.xyz' \
--header 'Content-Type: application/json' \
--data '{
    "jsonrpc": "2.0",
    "id": 1,
    "method": "getTransaction",
    "params": [
     "32wNq1e37Azb7K81qXBVodmpfZKuKxPpuYuBYaXt6SD2j5g4cNhLkk2z3zaZX9c486T82CWMkMHGjZ6esdyyYWWJ",
      "json"
    ]
}' | jq .result.meta.logMessages
```

Reply:
However when parsing logs, we need to make sure to drop any logs that occur before the `invoke` log of the outbox, and after the outbox returns (likely the `Dispatched message log`). Otherwise someone could create a contract that logs `"Protocol fee of {} paid from {} to {}"` and then invokes `outbox_dispatch`, and the scraper wouldn't know which log is the correct one

Reply:
yeah good call. I recall there being an edge case with logs fwiw https://github.com/coral-xyz/anchor/issues/1613 but I think this is fine as I can't imagine the incentive for someone to not have protocol fees indexed

Reply:
interesting workaround is to make a self-CPI with call data as the log, since that's never truncated / removed
https://book.anchor-lang.com/anchor_in_depth/events.html#cpi-events